tests/Functional/RegionRenderTest.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace Drupal\Tests\block_content\Functional;
+
+use Drupal\Tests\BrowserTestBase;
+
+/**
+ * Tests the code in localgov_base_preprocess_page() that renders regions.
+ *
+ * @group localgov_base
+ */
+class RegionRenderTest extends BrowserTestBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $modules = [
+    'localgov_base_test_support',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected $profile = 'localgov';
+
+  /**
+   * {@inheritdoc}
+   */
+  protected $defaultTheme = 'localgov_base';
+
+  /**
+   * Data provider for ::testRegionRender().
+   */
+  public function blockRegionProvider() {
+    return [
+      [
+        'region' => 'disabled',
+        'rendered' => 'FALSE',
+      ],
+      [
+        'region' => 'sidebar_first',
+        'rendered' => 'TRUE',
+      ],
+    ];
+  }
+
+  /**
+   * Tests that blocks in regions that aren't printed don't get rendered.
+   *
+   * To do this, we place a block provided by the localgov_base_test_support
+   * module into one of the regions. We then load a page, which is a callback
+   * provided by a controller in the same module, that will always load, and
+   * show blocks. Our block class records if an instance of it was built when
+   * the page loads. The module registers an event listener to add if the block
+   * was rendered to the response headers.
+   *
+   * We do all this because the page render runs in a different process to this
+   * test, so we can't just read the value from the class directly.
+   *
+   * @dataProvider blockRegionProvider
```

Comment:
OK so this is testing if something is rendered and something isn't rendered. That's pretty much all we need to do here. And as each run is a separate test the static value setting the header will be different and correct.